src/results/ResultsContainer.tsx

```diff
@@ -118,25 +138,113 @@ export class Results extends Component<ResultsProps, ResultsContainerState> {
       prevSelectedRows[rowId] = true;
       this.setState({ selectedRows: prevSelectedRows });
     }
-  }
+  };
+
+  facetsAsString = (facets: SelectedFacet[]) => {
+    if (!facets || facets.length <= 0) {
+      return '';
+    }
+    return facets.reduce(
+      (accumulator, facet, i) =>
+        `${accumulator}${i > 0 ? ',' : ''}${facet.name}:${facet.value}`,
+      '&facets='
+    );
+  };
+
+  facetsAsArray = (facetString: string): SelectedFacet[] => {
+    return facetString.split(',').map(stringItem => {
+      const item = stringItem.split(':');
+      return {
+        name: item[0],
+        value: item[1],
+      };
+    });
+  };
+
+  addFacet = (facetName: string, facetValue: string) => {
+    const {
+      location: { search: queryParamFromUrl },
+    } = this.props;
+    const {
+      query,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+    } = this.getURLParams(queryParamFromUrl);
+
+    const facet: SelectedFacet = { name: facetName, value: facetValue };
+
+    this.setURLParams(
+      query,
+      [...selectedFacets.slice(0, selectedFacets.length), facet],
+      sortColumn,
+      sortDirection
+    );
+  };
+
+  removeFacet = (facetName: string, facetValue: string) => {
+    const {
+      location: { search: queryParamFromUrl },
+    } = this.props;
+    const {
+      query,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+    } = this.getURLParams(queryParamFromUrl);
+
+    const index = selectedFacets.findIndex(
+      selectedFacet =>
+        selectedFacet.name === facetName && selectedFacet.value === facetValue
+    );
+
+    this.setURLParams(
+      query,
+      [...selectedFacets.slice(0, index), ...selectedFacets.slice(index + 1)],
```

Comment:
To be more clear about what is this going on, this line also can be written as:

     // Before calling this.setURLParamas
     const updatedSelectedFacets = selectedFacets.splice(index, 1);
     // Within this.setURLParams
     updatedSelectedFacets,